hangman/src/App.js

```diff
@@ -0,0 +1,223 @@
+          hint: "I'm always in front of you, but I cannot be seen",
+        },
+        {
+          level: 6,
+          word: "Echo",
+          hint: "What can’t talk but will reply when spoken to",
+        },
+        {
+          level: 7,
+          word: "Yarn",
+          hint:
+            "I’m found in socks, scarves and mittens; and often in the paws of playful kittens",
+        },
+        {
+          level: 8,
+          word: "Window",
+          hint: "What invention lets you look right through a wall?",
+        },
+        {
+          level: 9,
+          word: "Secret",
+          hint:
+            "If you’ve got me, you mustn't share me; if you share me, you haven’t kept me. What am I?",
+        },
+        {
+          level: 10,
+          word: "Name",
+          hint:
+            "It belongs to you, but other people use it more than you do. What is it?",
+        },
+      ],
+      activeSolution: {},
+      score: 100,
+      gameStatus: {
+        isGameStarted: false,
+        isGameOver: false,
+        isPlayerWon: false,
+      },
+    }
+  }
+  generateLetterStatuses() {
+    let letterStatus = {}
+    for (let i = 65; i < 91; i++) {
+      letterStatus[String.fromCharCode(i)] = false
+    }
+    return letterStatus
+  }
+  resetStates = {
+    letterStatus: async () => {
+      let resetLetterStatus = this.generateLetterStatuses()
+      this.setState({ letterStatus: resetLetterStatus })
+    },
+    gameStatus: async () => {
+      let resetGameStatus = this.state.gameStatus
+      resetGameStatus.isGameOver = false
+      resetGameStatus.isPlayerWon = false
+      resetGameStatus.isGameStarted = true
+      this.setState({ gameStatus: resetGameStatus })
+    },
+    score: async () => {
+      this.setState({ score: 100 })
+    },
+  }
+  findSolutionOfLevel = (level) => {
+    const solution = this.state.solutions.find((s) => s.level === level)
+    return solution
+  }
+  endGame = () => {
+    let updatedGameStatus = this.state.gameStatus
+    updatedGameStatus.isGameOver = true
+    updatedGameStatus.isPlayerWon = this.state.score ? true : false
+    this.setState({ gameStatus: updatedGameStatus })
+  }
+  selectLetter = (letter) => {
+    let allLetters = { ...this.state.letterStatus }
+    allLetters[letter] = true
+    this.setState(
+      {
+        letterStatus: allLetters,
+      },
+      () => {
+        const word = this.findSolutionOfLevel(this.state.activeSolution.level)
+          .word
+        let isLetterInWord =
+          word.includes(letter) || word.includes(letter.toLowerCase())
+        if (isLetterInWord) {
+          let updatedSolution = this.state.activeSolution
+          updatedSolution.lettersToFind--
+          this.setState({ activeSolution: updatedSolution }, () => {
+            if (this.state.activeSolution.lettersToFind === 0) {
+              if (
+                this.state.activeSolution.level === this.state.solutions.length
+              ) {
```

Comment:
You aren't using the `letterStatus` state inside the setState callback. so there is no reason to put all of this logic in the callback. you can do all of this in one setState.